addons/bus/static/tests/mock_server/mock_models/bus_bus.js

```diff
@@ -67,8 +67,7 @@ export class BusBus extends models.Model {
         const authenticatedUserId =
             "res.users" in this.env && this.env.cookie.get("authenticated_user_sid");
         const channels = [
-            ...IrWebSocket._build_bus_channel_list(),
-            ...(this.channelsByUser[authenticatedUserId] || []),
+            ...IrWebSocket._build_bus_channel_list(this.channelsByUser[authenticatedUserId] || []),
         ];
         notifications = notifications.filter(([target]) =>
```

Comment:
Looks like this diff only affects order: `channelsByUser` was after `"broadcast"` and a `res.partner`, now it's before. Is the ordering important for the fix?

Reply:
It's not related to order. Non members can access channels. In this case, the user subscribes to the `discuss.channel_${CHANNEL_ID}` string channel. Then, the `_build_bus_channel_list` method removes this string from the channels and replace it by the channel record. Before this change, the user channels were not passed to the `_build_bus_channel_list` method thus preventing the subscription to the channel notifications.